tests/functional/test_charm.py

```diff
@@ -533,6 +533,14 @@ async def test_resource_clean_up(self, ops_test, app, unit, required_resources):
 class TestCharm:
     """Perform basic functional testing of the charm without having the actual hardware."""
 
+    async def test_config_file_permissions(self, unit, ops_test):
+        """Check config file permissions are set correctly."""
+        expected_file_mode = "600"
+        cmd = "stat -c '%a' /etc/hardware-exporter-config.yaml"
```

Comment:
This may fix the functional tests
```suggestion
        cmd = "stat --printf='%a' /etc/hardware-exporter-config.yaml"
```

Reply:
I faced the same issue even with the `printf` flag. So I resorted to stripping out the newline from the result instead.